app/controllers/posts_controller.rb

```diff
@@ -1,23 +1,42 @@
 class PostsController < ApplicationController
-    def create
-        Post.create(post_params)
-        redirect_to root_path
-    end
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  def index
+    if current_user
+        @posts = Post.paginate(page: params[:page], per_page: 15).order('created_at DESC')
+    else
+        redirect_to new_user_session_path
+    end  
+  end
 
-    def show
-        @post = Post.find(params[:id])
-    end
+  def create
+    Post.create(post_params)
+    redirect_to root_path
+  end
 
-    def destroy
-        @post = current_user.posts.find(params[:id])
-        @post.destroy
+  def show
+  end
 
-        redirect_to user_path(current_user)
-    end
+  def edit
+  end
 
-    private
+  def update
+    @post.update post_params
+    redirect_to @post
+  end
 
-    def post_params
-        params.require(:post).permit(:description, :image, :user_id)
-    end
+  def destroy
+    @post = current_user.posts.find(params[:id])
+    @post.destroy
+
+    redirect_to root_path
+  end
+
+  private
+
+  def set_post
+    @post  = Post.find(params[:id])
```

Comment:
what happens if there is not post with id passed in params,

Reply:
For this, I changed find to find_by. Now, I'm redirecting to root with some error msg for this case